From: Ricardo Salveti de Araujo <ricardo.salveti@canonical.com>
Date: Wed, 10 Apr 2013 03:16:03 -0300
Subject: [PATCH] Revert "core: don't enumerate uninitialized devices from udev
 (fdo #56929)"

This reverts commit c6fb063e0906469beb9c0bbf214052d0d39a145d as it
breaks compatibility with Ubuntu Touch (lack of a fully working udev).

Signed-off-by: Ricardo Salveti de Araujo <ricardo.salveti@canonical.com>
---
 src/nm-udev-manager.c | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/src/nm-udev-manager.c b/src/nm-udev-manager.c
index de657e9..018460d 100644
--- a/src/nm-udev-manager.c
+++ b/src/nm-udev-manager.c
@@ -520,35 +520,25 @@ void
 nm_udev_manager_query_devices (NMUdevManager *self)
 {
 	NMUdevManagerPrivate *priv = NM_UDEV_MANAGER_GET_PRIVATE (self);
-	GUdevEnumerator *enumerator;
 	GList *devices, *iter;
 
 	g_return_if_fail (self != NULL);
 	g_return_if_fail (NM_IS_UDEV_MANAGER (self));
 
-	enumerator = g_udev_enumerator_new (priv->client);
-	g_udev_enumerator_add_match_subsystem (enumerator, "net");
-	g_udev_enumerator_add_match_is_initialized (enumerator);
-
-	devices = g_udev_enumerator_execute (enumerator);
+	devices = g_udev_client_query_by_subsystem (priv->client, "net");
 	for (iter = devices; iter; iter = g_list_next (iter)) {
 		net_add (self, G_UDEV_DEVICE (iter->data));
 		g_object_unref (G_UDEV_DEVICE (iter->data));
 	}
 	g_list_free (devices);
-	g_object_unref (enumerator);
 
 
-	enumerator = g_udev_enumerator_new (priv->client);
-	g_udev_enumerator_add_match_subsystem (enumerator, "atm");
-	g_udev_enumerator_add_match_is_initialized (enumerator);
-	devices = g_udev_enumerator_execute (enumerator);
+	devices = g_udev_client_query_by_subsystem (priv->client, "atm");
 	for (iter = devices; iter; iter = g_list_next (iter)) {
 		adsl_add (self, G_UDEV_DEVICE (iter->data));
 		g_object_unref (G_UDEV_DEVICE (iter->data));
 	}
 	g_list_free (devices);
-	g_object_unref (enumerator);
 }
 
 static void
-- 
1.8.1.2

